vernac/declare.ml

```diff
@@ -486,16 +496,19 @@ let inVariable v = Libobject.Dyn.Easy.inj v objVariable
 let warn_opaque_let = CWarnings.create ~name:"opaque-let" ~category:"fragile"
   Pp.(fun name ->
     Id.print name ++
-    strbrk " is declared opaque but this is not fully respected" ++
-    strbrk " inside the section and not at all outside the section.")
+    strbrk " is declared opaque (Qed) but this is not fully respected" ++
+    strbrk " inside the section and not at all outside the section." ++ fnl() ++
+    strbrk "Use attribute #[clearbody] to get the current behaviour of clearing" ++
+    strbrk " the body at the start of proofs in a forward compatible way.")
 
-let declare_variable_core ~name ~kind d =
+let declare_variable_core ~name ~clearbody ~kind d =
```

Comment:
I'm unsure what the parameter `clearbody` brings here. It seems better to me just to respect `opaque`.

Reply:
declare_variable_core looks nicer now IMO